src/bundle/ui-dev/src/modules/universal-discovery/content.meta.preview.module.js

```diff
@@ -94,50 +98,72 @@ const ContentMetaPreview = () => {
             </div>
         );
     };
-    const lastModifiedLabel = Translator.trans(/*@Desc("Last modified")*/ 'meta_preview.last_modified', {}, 'universal_discovery_widget');
-    const creationDateLabel = Translator.trans(/*@Desc("Created")*/ 'meta_preview.creation_date', {}, 'universal_discovery_widget');
-    const translationsLabel = Translator.trans(/*@Desc("Translations")*/ 'meta_preview.translations', {}, 'universal_discovery_widget');
+    const renderMetaPreviewLoadingSpinner = () => {
+        const isSpinnerVisible = !locationData || !locationData.location || !locationData.version;
+        const spinnerClassName = createCssClassNames({
+            'c-content-meta-preview__loading-spinner': true,
+            'c-content-meta-preview__loading-spinner--hidden': !isSpinnerVisible
+        });
 
-    return (
-        <div className="c-content-meta-preview" ref={refContentMetaPreview}>
-            <div className="c-content-meta-preview__preview">
-                <Thumbnail thumbnailData={version.Thumbnail} iconExtraClasses="ibexa-icon--extra-large" />
-            </div>
-            {renderActions()}
-            <div className="c-content-meta-preview__header">
-                <span className="c-content-meta-preview__content-name">{location.ContentInfo.Content.TranslatedName}</span>
+        return (
+            <div className={spinnerClassName}>
+                <Icon name="spinner" extraClasses="ibexa-icon--medium ibexa-spin" />
             </div>
-            <div className="c-content-meta-preview__info">
-                <div className="c-content-meta-preview__content-type-name">
-                    {contentTypesMap[location.ContentInfo.Content.ContentType._href].name}
+        );
+    };
+    const renderMetaPreview = () => {
+        if (!locationData || !locationData.location || !locationData.version) {
```

Comment:
maybe some additional func or move it above, as you used this kind of statement before (line 102)